allennlp/commands/diff.py

```diff
@@ -0,0 +1,300 @@
+"""
+# Examples
+
+```bash
+allennlp diff \
+    https://huggingface.co/roberta-large/resolve/main/pytorch_model.bin \
+    https://storage.googleapis.com/allennlp-public-models/transformer-qa-2020-10-03.tar.gz!weights.th \
+    --strip-prefix-1 'roberta.' \
+    --strip-prefix-2 '_text_field_embedder.token_embedder_tokens.transformer_model.'
+```
+"""
+from collections import OrderedDict
+import argparse
+import logging
+from os import PathLike
+import re
+from typing import Union, Optional, Dict, List, Tuple, NamedTuple, cast
+
+from overrides import overrides
+import termcolor
+import torch
+
+from allennlp.commands.subcommand import Subcommand
+from allennlp.common.file_utils import cached_path
+
+
+logger = logging.getLogger(__name__)
+
+
+@Subcommand.register("diff")
+class Diff(Subcommand):
+    requires_plugins: bool = False
+
+    @overrides
+    def add_subparser(self, parser: argparse._SubParsersAction) -> argparse.ArgumentParser:
+        description = """Display a diff between two model checkpoints."""
+        subparser = parser.add_parser(
+            self.name,
+            description=description,
+            help=description,
+        )
+        subparser.set_defaults(func=_diff)
+        subparser.add_argument(
+            "checkpoint1",
+            type=str,
+            help="""The URL or path to the first PyTorch checkpoint file (e.g. '.pt' or '.bin').""",
+        )
+        subparser.add_argument(
+            "checkpoint2",
+            type=str,
+            help="""The URL or path to the second PyTorch checkpoint file.""",
+        )
+        subparser.add_argument(
+            "--strip-prefix-1",
+            type=str,
+            help="""A prefix to remove from all of the first checkpoint's keys.""",
+        )
+        subparser.add_argument(
+            "--strip-prefix-2",
+            type=str,
+            help="""A prefix to remove from all of the second checkpoint's keys.""",
+        )
+        return subparser
+
+
+def load_state_dict(
```

Comment:
Is this function not present anywhere else in the AllenNLP library? It might be better to import it than copy it here, to help with maintenance.

Reply:
Yes, good point. It should definitely be elsewhere.